packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,83 @@
-use std::path::PathBuf;
-
 use memory::shared_memory::MemoryId;
 
 use crate::{
-    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result},
+    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result, Error},
     proto::ExperimentId,
 };
 
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
```

Comment:
I ended up removing the `Panic` variant. The panic branch returns a "general error" and the binary would have to check the error message to see if a panic or another error. It didn't seem right.

Reply:
That's fair. If we want this behavior we can add this later.